pkg/apis/resource/validation/validation_resourceclaim_test.go

```diff
@@ -605,10 +610,206 @@ func TestValidateClaimStatusUpdate(t *testing.T) {
+		},
+		"invalid-device-status-duplicate": {
+			wantFailures: field.ErrorList{
+				field.Duplicate(field.NewPath("status", "devices").Index(1).Child("deviceID"), structured.DeviceID{Driver: goodName, Pool: goodName, Device: goodName}),
+			},
+			oldClaim: func() *resource.ResourceClaim { return validAllocatedClaim }(),
+			update: func(claim *resource.ResourceClaim) *resource.ResourceClaim {
+				claim = claim.DeepCopy()
+				claim.Status.Devices = []resource.AllocatedDeviceStatus{
+					{
+						Driver: goodName,
+						Pool:   goodName,
+						Device: goodName,
+					},
+					{
+						Driver: goodName,
+						Pool:   goodName,
+						Device: goodName,
+					},
+				}
+				return claim
+			},
+			deviceStatusFeatureGate: true,
+		},
+		"invalid-network-device-status": {
+			wantFailures: field.ErrorList{
+				field.Invalid(field.NewPath("status", "devices").Index(0).Child("networkData", "addresses").Index(0), "300.9.8.0/24", "must be a valid CIDR value, (e.g. 10.9.8.0/24 or 2001:db8::/64)"),
+			},
+			oldClaim: func() *resource.ResourceClaim { return validAllocatedClaim }(),
+			update: func(claim *resource.ResourceClaim) *resource.ResourceClaim {
+				claim = claim.DeepCopy()
+				claim.Status.Devices = []resource.AllocatedDeviceStatus{
+					{
+						Driver: goodName,
+						Pool:   goodName,
+						Device: goodName,
+						NetworkData: &resource.NetworkDeviceData{
+							Addresses: []string{
+								"300.9.8.0/24",
+							},
+						},
+					},
+				}
+				return claim
+			},
+			deviceStatusFeatureGate: true,
+		},
+		"invalid-data-device-status": {
+			wantFailures: field.ErrorList{
+				field.Invalid(field.NewPath("status", "devices").Index(0).Child("data"), "<value omitted>", "error parsing data: invalid character 'o' in literal false (expecting 'a')"),
+			},
+			oldClaim: func() *resource.ResourceClaim { return validAllocatedClaim }(),
+			update: func(claim *resource.ResourceClaim) *resource.ResourceClaim {
+				claim = claim.DeepCopy()
+				claim.Status.Devices = []resource.AllocatedDeviceStatus{
+					{
+						Driver: goodName,
+						Pool:   goodName,
+						Device: goodName,
+						Data: runtime.RawExtension{
+							Raw: []byte(`foo`),
+						},
+					},
+				}
+				return claim
+			},
+			deviceStatusFeatureGate: true,
+		},
+		"invalid-device-status-no-device": {
+			wantFailures: field.ErrorList{
+				field.Invalid(field.NewPath("status", "devices").Index(0), structured.DeviceID{Driver: "b", Pool: "a", Device: "r"}, "must be an allocated device in the claim"),
+			},
+			oldClaim: func() *resource.ResourceClaim { return validAllocatedClaim }(),
+			update: func(claim *resource.ResourceClaim) *resource.ResourceClaim {
+				claim = claim.DeepCopy()
+				claim.Status.Devices = []resource.AllocatedDeviceStatus{
+					{
+						Driver: "b",
+						Pool:   "a",
+						Device: "r",
+					},
+				}
+				return claim
+			},
+			deviceStatusFeatureGate: true,
+		},
+		"invalid-device-status-duplicate-disabled-feature-gate": {
```

Comment:
Is this the behavior we want? If the user forgets to enable the feature gate, will we simply accept updates to the `Status` object for the newly added `Devices` object, and not validate it?

Reply:
cc @pohly as it looks like he's been maintaining the `dropDisabledFields` functionality

Reply:
This has to follow the rules for changing APIs https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version

If the feature is disabled the fields are dropped so don't reach validation

Reply:
ACK

Reply:
I misunderstood your comment @jackfrancis , sorry,  I think you are correct, the validation should not be feture gated... the fields are feature gated , but old data should be allowed independently of the feature fate so this should fail IIRC